Updated `SnapshotTesting` to `1.10.0`

NachoSoto opened the pull request:
[The new version](https://github.com/pointfreeco/swift-snapshot-testing/releases/tag/1.10.0) requires `SPM`, so this also moves it away from the `Podfile`.

tonidero commented on ios/PurchasesHybridCommon/PurchasesHybridCommon.xcodeproj/project.pbxproj:
```diff
@@ -1299,6 +1307,25 @@
+			repositoryURL = "https://github.com/pointfreeco/swift-snapshot-testing";
+			requirement = {
+				kind = upToNextMajorVersion;
+				minimumVersion = 1.9.0;
```
I guess we don't have tests with < iOS 13 in PHC right?

NachoSoto replied:
Good point, yeah we don't. I think since we do test the underlying SDK in iOS 12+ it's fine if the PHC is only tested in one version.

tonidero reviewed (approved):
LGTM!

NachoSoto commented:
> Package.resolved file is corrupted or malformed; fix or delete the file to continue: unsupported schema version 2

Oh let me upgrade CI to Xcode 14.

NachoSoto commented:
I updated to Xcode 14, but now `watchOS` doesn't build when doing `pod lib lint`:
>     - NOTE  | [watchOS] xcodebuild:  xcodebuild: error: Unable to find a destination matching the provided destination specifier:

I fixed the supported families:
![Screenshot 2022-09-23 at 13 54 21](https://user-images.githubusercontent.com/685609/192054978-56a8b318-5b1d-4442-a3ae-0d03c58fe792.png)

But for some reason the autogenerated framework from the `Podfile` has the wrong families, so now it doesn't compile for `watchOS`:
![Screenshot 2022-09-23 at 13 54 49](https://user-images.githubusercontent.com/685609/192055041-8bc9aca9-b4b4-4866-a3f9-7aeca0d5f682.png)

NachoSoto commented:
Fixing that here: #232
